### ISSUE TITLE
Comparison operators not working in PHP 8.3

### ISSUE THREAD

**Issue by jakespencer** [php-decimal/ext-decimal#77]
**Code**

```php
<?php

var_dump(new \Decimal\Decimal(0) == 0);
var_dump(new \Decimal\Decimal(0) > 0);
```

**Expected Output**
```
bool(true)
bool(false)
```

**Actual Output**
```
PHP Notice:  Object of class Decimal\Decimal could not be converted to int in /app/issue.php on line 3

Notice: Object of class Decimal\Decimal could not be converted to int in /app/issue.php on line 3
bool(false)
PHP Notice:  Object of class Decimal\Decimal could not be converted to int in /app/issue.php on line 4

Notice: Object of class Decimal\Decimal could not be converted to int in /app/issue.php on line 4
bool(true)
```

**Versions**

php version: 8.3.0
decimal version: 1.4.0
libmpdec version: 2.5.1

**Comment by speller** [comment]
I have this issue too. This is a critical issue under PHP 8.3 that blocks the migration to the new PHP version.

**Comment by speller** [comment]
No only comparison operators, math operators also not working on PHP 8.3:

> TypeError : Unsupported operand types: Decimal\Decimal + Decimal\Decimal

**Comment by Kezino** [comment]
+1

**Comment by rtheunissen** [comment]
Hi everyone, I'll do my best to fix this ASAP. If anyone is seriously blocked and would like to take a look at changes to php internals, I'm sure the patch won't be significant.

**Comment by Majkl578** [comment]
Here's a PR to fix PHP 8.3 compatibility: #80

I'd appreciate if anyone can test this independently with their code base. 🙂

**Comment by bobvandevijver** [comment]
@Majkl578 Already merged, but I am happy to confirm your change indeed fixes the issue for me 🚀

**Comment by rtheunissen** [comment]
Released as 1.5.0 on PECL: https://pecl.php.net/package/decimal/1.5.0

**Comment by Kezino** [comment]
@rtheunissen , thank you very much! Great job!
